lib/msal-common/docs/Accounts.md

```diff
@@ -2,61 +2,49 @@
 
 MSAL Javascript libraries (`msal-browser` and `msal-node`) support both single account and multiple accounts scenarios in javascript applications. An `account` object is standardized across the libraries:
 
-```javascript
+```typescript
 export type AccountInfo = {
     homeAccountId: string;
     environment: string;
     tenantId: string;
     username: string;
+    localAccountId: string;
+    name?: string;
 };
 ```
 
-## Usage
-
-* We provide a public API `getAllAccounts()` which lists all the accounts currently in the cache. An application must choose an account to acquire tokens silently.
-* `msal-browser` provides a public API `getAccountByUsername()` for an application to retrieve a specific account based on `username`
-
-Sample usage is as below:
-
-``` javascript
-
-function handleResponse(resp) {
-    if (resp !== null) {
-        username = resp.account.username;
-        ...
-    } else {
-        // need to call getAccount here?
-        const currentAccounts = myMSALObj.getAllAccounts();
-        if (currentAccounts === null) {
-            return;
-        } else if (currentAccounts.length > 1) {
-            // Add choose account code here
-        } else if (currentAccounts.length === 1) {
-            // Single Account usecase
-            username = currentAccounts[0].username;
-            ...
-        }
-    }
-}
-
-async function getTokenPopup(request, account) {
-    request.account = account;
-    return await myMSALObj.acquireTokenSilent(request).catch(async (error) => {
-        console.log("silent token acquisition fails.");
-        if (error instanceof msal.InteractionRequiredAuthError) {
-            console.log("acquiring token using popup");
-            return myMSALObj.acquireTokenPopup(request).catch(error => {
-                console.error(error);
-            });
-        } else {
-            console.error(error);
-        }
-    });
-}
-```
+## Account Identifiers
+
+The following `AccountInfo` attributes are used identify user accounts in authentication contexts.
+
+### homeAccountId
+
+When MSAL obtains an authentication response, it checks if the response includes client information. Specifically, MSAL checks for the presence of:
+
+* `tenantId` - Unique identifier of the tenant the client application belongs to.
+* `uniqueId` - Unique identifier of the user account within the corresponding tenant.
+
+When these two attributes are present, the `homeAccountId` attribute is built by concatenating them the dot-separated format `uniqueId.tenantId`.
+
+In cases where there is no `tenantId` in the authentication response, such as when using `ADFS`, MSAL looks for the ID Token claim `sub`, which identifies the "subject" the ID Token makes claims about and, if present, sets it as the `homeAccountIdentifier`.
```

Comment:
Correction: `oid` is set in case of absence of `tid`. `sub` is only when `oid` claim is absent in `idToken`.

Reply:
usually it is `oid.tid` where `tid` is the homeTenant.

Reply:
As discussed offline, we'll merge like this for now since it reflects the state of the code but will be investigating to make sure this is technically correct, updating docs and code if necessary.